users/views.py

```diff
@@ -40,6 +42,16 @@ def get_object(self):
         # NOTE: this may be a logged in or anonymous user
         return self.request.user
 
+    def update(self, request, *args, **kwargs):
+        with transaction.atomic():
+            user_name = request.user.name
+            user_update_response = super(CurrentUserRetrieveUpdateViewSet, self).update(
+                request, *args, **kwargs
+            )
+            if user_name != request.data.get("name"):
+                api.update_edx_user(request.user)
```

Comment:
Can you add a new task to `courseware/tasks.py` and call it here? That task would call `api.update_edx_user`. The API call to edX might take some time, and we don't need to report an error on the user's end if the edX update request fails. It's best to run that as part of an asynchronous task. You can just pass the user id as an argument to the task

Reply:
Suggested changes have been incorporated along with a test.